element-template-generator/core/src/main/java/io/camunda/connector/generator/java/processor/TemplatePropertyFieldProcessor.java

```diff
@@ -190,6 +191,24 @@ private PropertyConstraints buildConstraints(
           new PropertyConstraints.Pattern(
               constraintsAnnotation.pattern().value(), constraintsAnnotation.pattern().message()));
     }
+    if (!constraintsAnnotation.secretPattern().value().isBlank()) {
+      if (!constraintsAnnotation.notEmpty() && propertyAnnotation.optional()) {
+        builder.notEmpty(false);
+      }
+      builder.pattern(
+          new PropertyConstraints.Pattern(
+              createSecretPatternValue(constraintsAnnotation.secretPattern().value(), property),
+              constraintsAnnotation.secretPattern().message()));
+    }
     return builder.build();
   }
+
+  private String createSecretPatternValue(String value, Property property) {
+    if (property.getFeel() == Property.FeelMode.optional
+        || property.getFeel() == Property.FeelMode.required) {
+      return "^(=|(secrets|%s)).*$".formatted(value.replaceAll("\\^", "").replaceAll("\\$", ""));
+    } else {
+      return "^(secrets|%s).*$".formatted(value.replaceAll("\\^", "").replaceAll("\\$", ""));
```

Comment:
do we need the `.*` at the end? Or should we just use the initial regex ( `%s`) ?

Reply:
Would you prefer something like 

`"^(secrets.*|(%s).*)$".formatted(value.replaceAll("\\^", "").replaceAll("\\$", ""));`

?